src/binary/ibc-setup/commands/balances.ts

```diff
@@ -0,0 +1,86 @@
+import os from 'os';
+import path from 'path';
+
+import { stringToPath } from '@cosmjs/crypto';
+import { GasPrice } from '@cosmjs/launchpad';
+import { DirectSecp256k1HdWallet } from '@cosmjs/proto-signing';
+
+import { Coin } from '../../../codec/cosmos/base/v1beta1/coin';
+import { IbcClient } from '../../../lib/ibcclient';
+import { registryFile } from '../../constants';
+import { loadAndValidateApp } from '../utils/load-and-validate-app';
+import { loadAndValidateRegistry } from '../utils/load-and-validate-registry';
+import { resolveHomeOption } from '../utils/options/shared/resolve-home-option';
+import { resolveKeyFileOption } from '../utils/options/shared/resolve-key-file-option';
+import { resolveMnemonicOption } from '../utils/options/shared/resolve-mnemonic-option';
+
+import { Flags, getAddresses, Options } from './keys-list';
+
+export { Options }; // re-export for testing
+
+export async function balances(flags: Flags) {
+  const home = resolveHomeOption({ homeFlag: flags.home });
+  const app = loadAndValidateApp(home);
+  const keyFile = resolveKeyFileOption({ keyFileFlag: flags.keyFile, app });
+  const mnemonic = await resolveMnemonicOption({
+    interactiveFlag: flags.interactive,
+    mnemonicFlag: flags.mnemonic,
+    keyFile: keyFile,
+    app,
+  });
+
+  const options: Options = {
+    home,
+    mnemonic,
+  };
+
+  await run(options);
+}
+
+export async function run(options: Options) {
+  const registryFilePath = path.join(options.home, registryFile);
+  const registry = loadAndValidateRegistry(registryFilePath);
+
+  const addresses = await getAddresses(registry.chains, options.mnemonic);
+
+  const balances = (
+    await Promise.all(
+      addresses.map<Promise<[string, Coin]>>(async ([chain, data, address]) => {
+        const signer = await DirectSecp256k1HdWallet.fromMnemonic(
+          options.mnemonic,
+          stringToPath(data.hd_path),
+          data.prefix
+        );
+
+        const gasPrice = GasPrice.fromString(data.gas_price);
+
+        const client = await IbcClient.connectWithSigner(
+          data.rpc[0], // rpc[0] is guaranteed to be defined by registry validator
+          signer,
+          address,
+          {
+            prefix: data.prefix,
+            gasPrice,
+          }
+        );
+
+        const coin = await client.query.bank.unverified.balance(
+          address,
+          gasPrice.denom
+        );
+
+        return [chain, coin];
+      })
+    )
+  )
+    .filter(([, coin]) => coin.amount !== '0')
+    .map(([chain, coin]) => `${chain}: ${coin.amount}`)
```

Comment:
We need to print the `denom` here as well. Eg 
```ts
`${chain}: ${coin.amount}${coin.denom}`
```

Reply:
1ad4c2d